kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaErrors.java

```diff
@@ -130,8 +134,45 @@ public static KernelException columnInvariantsNotSupported() {
         return new KernelException(message);
     }
 
-    /* ------------------------ HELPER METHODS ----------------------------- */
+    public static KernelException unsupportedDataType(DataType dataType) {
+        return new KernelException("Kernel doesn't support writing data of type: " + dataType);
+    }
+
+    public static KernelException unsupportedPartitionDataType(String colName, DataType dataType) {
+        String msgT = "Kernel doesn't support writing data with partition column (%s) of type: %s";
```

Comment:
Minor nit: "Delta doesn't support partitioning by partition column (%s) with type: %s"
- This is unsupported by Delta spec right not Kernel
- More clear that the issue is we cannot _partition_ by the column not just writing

Reply:
It is possible that Kernel may not have support for it, but Delta has it. For example `timestamp_ntz`.

Reply:
Isn't this rejected in the preceding schema check though?

Do you think we might at some point support some data column type but not support it for partition columns? I feel like that scenario could have a more specific error message. But for the cases we have here currently the data type is probably explicitly prohibited via the Delta spec.